pkg/protocompat/time.go

```diff
@@ -28,72 +29,77 @@ var (
 // The range is from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z. By restricting to that
 // range, we ensure that we can convert to and from
 // [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt ) date strings.
-type Timestamp = gogoTimestamp.Timestamp
+type Timestamp = timestamppb.Timestamp
 
 // TimestampNow returns a protobuf timestamp set to the current time.
-func TimestampNow() *gogoTimestamp.Timestamp {
-	return gogoTimestamp.TimestampNow()
+func TimestampNow() *timestamppb.Timestamp {
+	return timestamppb.Now()
 }
 
 // ConvertTimestampToCSVString converts a proto timestamp to a string for display in a CSV report.
-func ConvertTimestampToCSVString(timestamp *gogoTimestamp.Timestamp) string {
+func ConvertTimestampToCSVString(timestamp *timestamppb.Timestamp) string {
 	if timestamp == nil {
 		return "N/A"
 	}
-	ts, err := gogoTimestamp.TimestampFromProto(timestamp)
-	if err != nil {
+	if timestamp.CheckValid() != nil {
 		return "ERR"
 	}
-	return ts.Format(time.RFC1123)
+	return timestamp.AsTime().Format(time.RFC1123)
 }
 
 // ConvertTimestampToTimeOrNil converts a proto timestamp to a golang Time, defaulting to nil in case of error.
-func ConvertTimestampToTimeOrNil(gogo *gogoTimestamp.Timestamp) *time.Time {
-	if gogo == nil {
+func ConvertTimestampToTimeOrNil(pbTime *timestamppb.Timestamp) *time.Time {
+	if pbTime == nil {
 		return nil
 	}
-	goTime, err := ConvertTimestampToTimeOrError(gogo)
+	goTime, err := ConvertTimestampToTimeOrError(pbTime)
 	if err != nil {
 		return nil
 	}
 	return &goTime
 }
 
 // ConvertTimeToTimestampOrNil converts a golang Time to a proto timestamp, defaulting to nil in case of error.
-func ConvertTimeToTimestampOrNil(goTime *time.Time) *gogoTimestamp.Timestamp {
+func ConvertTimeToTimestampOrNil(goTime *time.Time) *timestamppb.Timestamp {
 	if goTime == nil {
 		return nil
 	}
-	gogo, err := ConvertTimeToTimestampOrError(*goTime)
+	pbTime, err := ConvertTimeToTimestampOrError(*goTime)
 	if err != nil {
 		return nil
 	}
-	return gogo
+	return pbTime
 }
 
 // ConvertTimestampToGraphqlTimeOrError converts a proto timestamp
 // to a graphql Time, or returns an error if there is one.
-func ConvertTimestampToGraphqlTimeOrError(gogo *gogoTimestamp.Timestamp) (*graphql.Time, error) {
-	if gogo == nil {
+func ConvertTimestampToGraphqlTimeOrError(pbTime *timestamppb.Timestamp) (*graphql.Time, error) {
+	if pbTime == nil {
 		return nil, nil
 	}
-	t, err := gogoTimestamp.TimestampFromProto(gogo)
-	return &graphql.Time{Time: t}, err
+
+	// TODO: We didn't cover error case in unit tests!
```

Comment:
if you opened a tech debt ticket, Could you reference it in the TODO?

Reply:
This is already addressed with https://github.com/stackrox/stackrox/pull/11438. I forgot to remove the comment.